src/app/accounts/settings.component.html

```diff
@@ -7,14 +7,7 @@
                         {{'security' | i18n}}
                     </div>
                     <div class="box-content box-content-padded">
-                        <div class="form-group">
-                            <label for="vaultTimeouts">{{'vaultTimeout' | i18n}}</label>
-                            <select id="vaultTimeouts" name="VaultTimeouts" [(ngModel)]="vaultTimeout"
-                                (change)="saveVaultTimeoutOptions()">
-                                <option *ngFor="let o of vaultTimeouts" [ngValue]="o.value">{{o.name}}</option>
-                            </select>
-                            <small class="help-block">{{'vaultTimeoutDesc' | i18n}}</small>
-                        </div>
+                        <app-vault-timeout-input [vaultTimeouts]="vaultTimeouts" [formControl]="vaultTimeout" ngDefaultControl></app-vault-timeout-input>
```

Comment:
Reactive forms are fairly new to me, does this `[formControl]` directive automatically bind to the `form.vaultTimeout` value in the child component? (without using an explicit `@Input` decorator)

Reply:
@eliykat In other components we usually need to forward the ngModel bindings from the parent component to child, which requires the `@Input` directive.

The `VaultTimeoutInputComponent` instead implements the `ControlValueAccessor` which exposes the Angular form API. In this case the component combines the three inputs into a single timeout in minutes. It's not directly related to reactive forms and could also be used with `[ngModel]`.